simphony/io/file_mesh.py

```diff
@@ -0,0 +1,778 @@
+""" Mesh File
+
+This module contains the implentation to store, acces,
+and modify a file storing mesh data
+
+"""
+import tables
+import uuid
+
+from simphony.cuds.mesh import Point
+from simphony.cuds.mesh import Edge
+from simphony.cuds.mesh import Face
+from simphony.cuds.mesh import Cell
+
+from simphony.io.data_container_table import DataContainerTable
+
+MAX_POINTS_IN_EDGE = 2
+MAX_POINTS_IN_FACE = 4
+MAX_POINTS_IN_CELL = 8
+
+
+class _PointDescriptor(tables.IsDescription):
+    """ Descriptor for storing Point information
+
+    Provides the column definition to store Point
+    information (x, y, z).
+
+    """
+
+    uid = tables.StringCol(32, pos=0)
+    data = tables.StringCol(32, pos=1)
+    coordinates = tables.Float64Col(
+        pos=2, shape=(3,)
+        )
+
+
+class _EdgeDescriptor(tables.IsDescription):
+    """ Descriptor for storing Edge information
+
+    Provides the column definition to store Edges
+    forming an edge and its data.
+
+    """
+
+    uid = tables.StringCol(32, pos=0)
+    data = tables.StringCol(32, pos=1)
+    points_uids = tables.StringCol(
+        32, pos=2, shape=(MAX_POINTS_IN_EDGE,)
+        )
+    n_points = tables.UInt32Col(pos=3)
+
+
+class _FaceDescriptor(tables.IsDescription):
+    """ Descriptor for storing Face information
+
+    Provides the column definition to store Faces
+    forming a face and its data.
+
+    """
+
+    uid = tables.StringCol(32, pos=0)
+    data = tables.StringCol(32, pos=1)
+    points_uids = tables.StringCol(
+        32, pos=2, shape=(MAX_POINTS_IN_FACE,)
+        )
+    n_points = tables.UInt32Col(pos=3)
+
+
+class _CellDescriptor(tables.IsDescription):
+    """ Descriptor for storing Cell information
+
+    Provides the column definition to store Cells
+    forming a cell and its data.
+
+    """
+
+    uid = tables.StringCol(32, pos=0)
+    data = tables.StringCol(32, pos=1)
+    points_uids = tables.StringCol(
+        32, pos=2, shape=(MAX_POINTS_IN_CELL,)
+        )
+    n_points = tables.UInt32Col(pos=3)
+
+
+class FileMesh(object):
+    """ FileMesh.
+
+    Interface of the mesh file driver.
+    Stores general mesh information Points and Elements
+    such as Edges, Faces and Cells and provide the
+    methods to interact with them. The methods are
+    divided in four diferent blocks:
+
+    (1) methods to get the related item with the provided uid;
+    (2) methods to add a new item or replace;
+    (3) generator methods that return iterators
+        over all or some of the mesh items and;
+    (4) inspection methods to identify if there are any edges,
+        faces or cells described in the mesh.
+
+    Attributes
+    ----------
+    data : Data
+        The X coordinate.
+    points : dictionary of Point
+        Points of the mesh.
+    edges : dictionary of Edge
+        Edges of the mesh.
+    faces : dictionary of Face
+        Faces of the mesh.
+    cells : dictionary of Cell
+        Cells of the mesh.
```

Comment:
`name` should be added.  and `points`, `edges`, `faces` and `cells` can be removed.